rule_stop.go

```diff
@@ -20,8 +27,31 @@ func (slr stopLossRule) IsSatisfied(index int, record *TradingRecord) bool {
 	if !record.CurrentPosition().IsOpen() {
 		return false
 	}
-
-	openPrice := record.CurrentPosition().CostBasis()
+	openPrice := record.CurrentPosition().EntranceOrder().Price
```

Comment:
This looks like a breaking change. `EntranceOrder().Price` reflects the unit price you paid for a security, not the total value of the position. Even though I think what you're doing here isn't incorrect, it would be breaking change for anyone who was depending on this behavior.